ggshield/scan/repo.py

```diff
@@ -85,28 +88,20 @@ def scan_commits_content(
             ignore_known_secrets=ignore_known_secrets,
         )
         results = scanner.scan(
-            commit_files,
+            [file for _, file in commit_files_tuple],
             scan_threads=SCAN_THREADS,
         )
     except Exception as exc:
         results = Results.from_exception(exc)
 
     scans = []
-    for commit in commits:
-        concerned_results = [
-            res
-            for res in results.results
-            if any(
-                res.content == file.document and res.filename == file.filename
-                for file in commit.files
-            )
-        ]
+    for (commit, file), result in zip(commit_files_tuple, results.results):
```

Comment:
Looks like we do not use `file` in this loop, so it can be replaced with `_`:
```suggestion
    for (commit, _), result in zip(commit_files_tuple, results.results):
```

It's interesting that this new code is probably going to be faster! 😀

Reply:
@agateau-gg  
Side note : if I am not mistaken we do not benchmark `scan repo` in the performance benchmark ? We could have seen it here. But it may be too long to include the repo option in the benchmark ?

Reply:
We benchmark `scan commit-range`, which uses this code.